### DISCUSSION TITLE
Identifying a codespace's region

### DISCUSSION THREAD

**Discussion by CodingCanuck** [community/community#21246] [Codespaces] [answered]
How do I tell the region in which a codespace is running? I don't see region / location information under "Codespaces: Details" or the creation log, nor do I see any region information under https://github.com/codespaces .

My [codespaces settings](https://github.com/settings/codespaces) use the "Set automatically" region selection strategy: this means I'm not sure where my codespace was created, especially when I'm traveling.

I'm interested in my codespace's region because I'm trying to optimize my team's bazel remote build cache deployment: we'd like our cloud-hosted build cache storage to be hosted in regions that are close to our codespaces' regions to hopefully minimize build latency.

**Comment by jungaretti** [comment, marked as answer]
You can use the GitHub CLI to see your codespace's region. From within a codespace that uses our default image, run the command `gh api /user/codespaces/$CODESPACE_NAME --jq .location` to print the codespace's region.

<img width="707" alt="gh api /user/codespaces/$CODESPACE_NAME --jq .location" src="https://user-images.githubusercontent.com/19893438/180880970-6ee22c80-416c-4258-ae99-4c760dcace9c.png">
